cpp/cppsdk/gsdk.cpp

```diff
@@ -123,9 +124,11 @@ namespace Microsoft
                     m_transitionToActiveEvent.Reset();
                     m_signalHeartbeatEvent.Reset();
 
+                    std::string infoUrl = "http://" + gsmsBaseUrl + "v1/metrics/" + instanceId + "/gsdkinfo";
+
                     // we might not want to heartbeat in our UTs
                     m_keepHeartbeatRunning = config->shouldHeartbeat();
-                    m_heartbeatThread = std::thread(&GSDKInternal::heartbeatThreadFunc, this);
+                    std::thread(&GSDKInternal::start, this, infoUrl).detach();
```

Comment:
Instead of creating this detached thread which then creates `m_heartbeatThread`, what if you just made `m_heartbeatThread` call `GSDKInternal::start` here and then changed `GSDKInternal::start` to begin the heartbeat loop after it completes the send-info request? That way the thread model stays the same and we aren't creating an extra thread

Reply:
I'd considered that initially. At the moment I can't remember what reasoning led me to implement this instead. If I can't remember after looking at this further, I'd be happy to change to that